Search filters accessibility improvements

nikomakela opened the pull request:
HH-379.

Aria-labelledby property is used to track which component controls it's visibility.

Mark the filter summary area as live atomic area, so the screen readers will tell whether something changes there.

Add aria-labelledby properties to each filter button type.

nikomakela committed:
feat(search): add aria-labelledby to all filter status buttons

HH-379.

Aria-labelledby property is used to track which component controls it's
visibility.

nikomakela committed:
feat(apps,search): add accessibility proeprties to filter summary

HH-379.

Mark the filter summary area as live atomic area, so the screen readers
will tell whether something changes there.

Add aria-labelledby properties to each filter button type.

terovirtanen commented:
**FEDERATION-ROUTER branch is deployed to platta: https://events-graphql-federation-events-pr850.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**VENUE-GRAPHQL-PROXY branch is deployed to platta: https://venue-graphql-proxy-pr850.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**LIIKUNTA-UI branch is deployed to platta: https://liikunta-pr850.dev.hel.ninja** :rocket::rocket::rocket:

nikomakela committed:
refactor(search): use aria-describedby with age selectors

HH-379.

> The aria-describedby attribute lists the ids of the elements that
describe the object. It is used to establish a relationship between
widgets or groups and the text that describes them.

> The aria-describedby attribute is not limited to form controls. It can
also be used to associate static text with widgets, groups of elements,
regions that have a heading, definitions, and more. The aria-describedby
attribute can be used with semantic HTML elements and with elements that
have an ARIA role.

Reference:
https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Reference/
Attributes/aria-describedby.

terovirtanen commented:
**FEDERATION-ROUTER branch is deployed to platta: https://events-graphql-federation-events-pr850.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**FEDERATION-ROUTER branch is deployed to platta: https://events-graphql-federation-hobbies-pr850.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**TAPAHTUMAT-UI branch is deployed to platta: https://tapahtumat-pr850.dev.hel.ninja** :rocket::rocket::rocket:

karisal-anders reviewed (approved):
SonarCloud's `<output>` tag use instead of aria-live `<div>` could be considered.

terovirtanen commented:
**FEDERATION-ROUTER branch is deployed to platta: https://events-graphql-federation-events-pr850.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**FEDERATION-ROUTER branch is deployed to platta: https://events-graphql-federation-hobbies-pr850.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
## TestCafe result is __success__ for https://tapahtumat-pr850.dev.hel.ninja :laughing::tada::tada::tada:

nikomakela committed:
refactor(search): filter summary rendered as output element

HH-379.

The SonarCloud reported this as an issue: "Use <output> instead of the
'status' role to ensure accessibility across all devices".

Chromium AI prefers to add the `role="status"` anyway: "While the
<output> element has a semantic meaning in HTML5 (representing the
result of a calculation or user action), it doesn't automatically have
the role="status" or aria-live="polite" attributes applied by default."

MDN describes that many browsers will implement output-element as an
aria-live region -- reference:
https://developer.mozilla.org/en-US/docs/Web/HTML/Reference/Elements/
output#accessibility

terovirtanen commented:
**FEDERATION-ROUTER branch is deployed to platta: https://events-graphql-federation-hobbies-pr850.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
## TestCafe result is __success__ for https://tapahtumat-pr850.dev.hel.ninja :laughing::tada::tada::tada:

terovirtanen commented:
**VENUE-GRAPHQL-PROXY branch is deployed to platta: https://venue-graphql-proxy-pr850.dev.hel.ninja** :rocket::rocket::rocket:
